packages/react-core/src/core/providers/thirdweb-provider.tsx

```diff
@@ -0,0 +1,188 @@
+   */
+  queryClient?: QueryClient;
+
+  /**
+   * Whether or not to attempt auto-connect to a wallet.
+   */
+  autoConnect?: boolean;
+
+  // api keys that can be passed
+  thirdwebApiKey?: string;
+  alchemyApiKey?: string;
+  infuraApiKey?: string;
+
+  /**
+   * A partial map of chainIds to rpc urls to use for certain chains
+   * If not provided, will default to the rpcUrls of the chain objects for the supported chains
+   * @deprecated - use `chains` instead
+   */
+  chainRpc?: Record<number, string>;
+
+  theme?: "light" | "dark";
+
+  createWalletStorage: CreateAsyncStorage;
+}
+
+// SDK handles this under the hood for us
+
+const defaultdAppMeta: DAppMetaData = {
+  name: "thirdweb powered dApp",
+  url: "https://thirdweb.com",
+};
+
+export const ThirdwebProvider = <
+  TChains extends Chain[] = typeof defaultChains,
+>(
+  props: React.PropsWithChildren<ThirdwebProviderProps<TChains>>,
+) => {
+  // deprecations
+  if (props.chainRpc) {
+    showDeprecationWarning("chainRpc", "supportedChains");
+  }
+
+  const supportedChains =
+    props.supportedChains || (defaultChains as any as TChains);
+
+  const dAppMeta = props.dAppMeta || defaultdAppMeta;
+  const activeChainObj = useMemo(() => {
+    if (!props.activeChain) {
+      return supportedChains[0];
+    }
+    if (typeof props.activeChain === "number") {
+      return (
+        supportedChains.find((chain) => chain.chainId === props.activeChain) ||
+        supportedChains[0]
+      );
+    }
+    if (typeof props.activeChain === "string") {
+      return (
+        supportedChains.find((chain) => chain.slug === props.activeChain) ||
+        supportedChains[0]
+      );
+    }
+
+    return props.activeChain || supportedChains[0] || defaultChains[0];
+  }, [props.activeChain, supportedChains]);
+
+  return (
+    <ThirdwebThemeContext.Provider value={props.theme}>
+      <ThirdwebWalletProvider
+        chains={supportedChains}
+        supportedWallets={props.supportedWallets}
+        shouldAutoConnect={props.autoConnect}
+        createWalletStorage={props.createWalletStorage}
+        dAppMeta={dAppMeta}
+        activeChain={activeChainObj}
+      >
+        <ThirdwebSDKProviderWrapper
+          queryClient={props.queryClient}
+          sdkOptions={props.sdkOptions}
+          supportedChains={supportedChains}
+          activeChain={activeChainObj.chainId}
+          storageInterface={props.storageInterface}
+          authConfig={props.authConfig}
+          thirdwebApiKey={props.thirdwebApiKey}
+          alchemyApiKey={props.alchemyApiKey}
+          infuraApiKey={props.infuraApiKey}
+        >
+          {props.children}
+        </ThirdwebSDKProviderWrapper>
+      </ThirdwebWalletProvider>
+    </ThirdwebThemeContext.Provider>
+  );
+};
+
+const ThirdwebSDKProviderWrapper = <TChains extends Chain[]>({
+  children,
+  ...props
+}: React.PropsWithChildren<
+  Omit<ThirdwebSDKProviderProps<TChains>, "signer">
+>) => {
+  const signer = useWalletSigner();
```

Comment:
can this hook just be called `useSigner()`?

Reply:
We already have a `useSigner()` hook. This is a duplicate and only used one-time

`useWalletSigner()` is used to provider the signer to `ThirdwebConnectedWalletContext` and `useSigner()` returns the signer from `ThirdwebConnectedWalletContext`. 

`useSigner()` is the one that is exported. `useSigner()` works in both cases - when app is wrapped with `<ThirdwebProvider/>` or `<ThirdwebSDKProvider/>`. but `useWalletSigner()` only works if the app is wrapped with `<ThirdwebProvider/>`

So What I can do it remove the `useWalletSigner()` hook and just use `useThirdwebWallet().signer` at a single place this is being used: 

<img width="639" alt="image" src="https://user-images.githubusercontent.com/22043396/224300553-a91779eb-852d-4f06-b0d9-c80db1862bc6.png">

Reply:
removed useWalletSigner()